# Description: Chromium has a common blacklist for GPU rasterization and
#  accelerated canvas on Android, as they use the same code paths. Enable
#  just accelerated canvas on Krillin though, as this is known to work well
#  enough. We keep GPU rasterization disabled, as this is known to not work
#  well at all on this device
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/gpu/config/software_rendering_list_json.cc b/gpu/config/software_rendering_list_json.cc
--- a/gpu/config/software_rendering_list_json.cc
+++ b/gpu/config/software_rendering_list_json.cc
@@ -1014,17 +1014,17 @@ LONG_STRING_CONST(
         "value": "13.101"
       },
       "features": [
         "all"
       ]
     },
     {
       "id": 96,
-      "description": "Blacklist GPU raster/canvas on all except known good GPUs and newer Android releases",
+      "description": "Blacklist GPU raster on all except known good GPUs and newer Android releases",
       "cr_bugs": [362779,424970],
       "os": {
         "type": "android"
       },
       "exceptions": [
         {
           "os": {
             "type": "android"
@@ -1069,17 +1069,82 @@ LONG_STRING_CONST(
           "gl_type": "gles",
           "gl_version": {
             "op": ">=",
             "value": "3.0"
           }
         }
       ],
       "features": [
-        "gpu_rasterization",
+        "gpu_rasterization"
+      ]
+    },
+    {
+      "id": 97,
+      "description": "Blacklist GPU canvas on all except known good GPUs and newer Android releases",
+      "cr_bugs": [362779,424970],
+      "os": {
+        "type": "android"
+      },
+      "exceptions": [
+        {
+          "os": {
+            "type": "android"
+          },
+          "gl_renderer": "Adreno (TM) 3.*"
+        },
+        {
+          "os": {
+            "type": "android"
+          },
+          "gl_renderer": "NVIDIA.*"
+        },
+        {
+          "os": {
+            "type": "android"
+          },
+          "gl_renderer": "VideoCore IV.*"
+        },
+        {
+          "os": {
+            "type": "android"
+          },
+          "gl_renderer": "Immersion.*"
+        },
+        {
+          "os": {
+            "type": "android"
+          },
+          "gl_renderer": "Mali-400.*"
+        },
+        {
+          "os": {
+            "type": "android",
+            "version": {
+              "op": ">=",
+              "value": "4.4.99"
+            }
+          }
+        },
+        {
+          "os": {
+            "type": "android",
+            "version": {
+              "op": ">=",
+              "value": "4.4"
+            }
+          },
+          "gl_type": "gles",
+          "gl_version": {
+            "op": ">=",
+            "value": "3.0"
+          }
+        }
+      ],
+      "features": [
         "accelerated_2d_canvas"
       ]
     },
     {
       "id": 99,
       "description": "GPU rasterization is blacklisted on non-Android",
       "cr_bugs": [362779],
       "exceptions": [
